nats.go

```diff
@@ -2692,7 +2695,11 @@ func (nc *Conn) processMsg(data []byte) {
 	}
 
 	// Copy them into string
-	subj := string(nc.ps.ma.subject)
+	subj := sub.Subject
+	jsi := sub.jsi
```

Comment:
Yeah, but this is not entirely immutable (there is a case where sub.jsi is set to nil when a js subscription fails at some point), so you may or may not get race reports here...

Reply:
Forgot to mention that where we had captured it down below, it was under the subscription lock... but now you move it outside of it.

Reply:
Actually even sub.Subject can mutate (for a JS subscription in resetOrderedConsumer). Hum... this is getting a bit out of hand, not sure if this is all worth the effort. Have you actually measured how much we gain? (with a go bench report)?

Reply:
I guess in this case 
sub.mu.Lock()

// Check if closed.
if sub.closed {
  sub.mu.Unlock()
  return
}
shows that subscription closed?

Reply:
I have an another idea

Reply:
But that's not the issue, the issue is that you access `sub.Subject` and `sub.jsi` outside of the subscription's lock while in some other go routines, those values may be changed at the same time (under the sub's lock). So you may get race reports.

Reply:
Check latest changes